src/ServiceControl/Operations/UnitOfWork/InMemory/InMemoryIngestionUnitOfWorkFactory.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace ServiceControl.Operations
+{
+    using System.Threading.Tasks;
+    using Monitoring;
+
+    class InMemoryIngestionUnitOfWorkFactory : IIngestionUnitOfWorkFactory
+    {
+        IMonitoringDataStore dataStore;
+
+        public InMemoryIngestionUnitOfWorkFactory(IMonitoringDataStore dataStore) => this.dataStore = dataStore;
+
+        public Task<IIngestionUnitOfWork> StartNew()
+            => Task.FromResult(
+                (IIngestionUnitOfWork)new InMemoryIngestionUnitOfWork(dataStore)
```

Comment:
Is this cast necessary since the class implements the interface and it is implied?

Reply:
I get a compiler warning about not being able to cast `Task<InMemoryIngestionUnitOfWork>` to `Task<IIngestionUnitOfWork>` without it.

Reply:
I turned this into `Task.FromResult<IIngestionUnitOfWork>(new InMemoryIngestionUnitOfWork(dataStore))` which seems neater